bcipy/helpers/report.py

```diff
@@ -0,0 +1,220 @@
+import io
+from abc import ABC
+from typing import List, Optional
+
+from matplotlib.figure import Figure
+from reportlab.platypus import SimpleDocTemplate, Paragraph, Image
+from reportlab.lib.pagesizes import letter
+from reportlab.lib.styles import getSampleStyleSheet
+from reportlab.platypus import Flowable, KeepTogether
+from reportlab.lib.units import inch
+
+from bcipy.config import BCIPY_FULL_LOGO_PATH
+
+
+class ReportSection(ABC):
+    """Report Section.
+
+    An abstract class to handle the creation of a section in a BciPy Report.
+    """
+
+    def _compile(self) -> Flowable:
+        ...
+
+    def _create_header(self) -> Flowable:
+        ...
+
+
+class SignalReport(ReportSection):
```

Comment:
Naming is always hard. It would be more consistent to name this `SignalReportSection`, since it subclasses `ReportSection` rather than `Report`. Alternatively we could rename `Report` to `ReportBuilder` or something like that.